galsim/compound.py

```diff
@@ -322,18 +322,26 @@ def __init__(self, *args, **kwargs):
             else:
                 real_space = False
 
-        # Warn if doing DFT convolution for objects with hard edges.
+        # Warn if doing DFT convolution for objects with hard edges
         if not real_space and hard_edge:
-            import warnings
-            if len(args) == 2:
-                msg = """
-                Doing convolution of 2 objects, both with hard edges.
-                This might be more accurate and/or faster using real_space=True"""
-            else:
-                msg = """
-                Doing convolution where all objects have hard edges.
-                There might be some inaccuracies due to ringing in k-space."""
-            warnings.warn(msg)
+            
+            # If any aren't analytic, we don't have a choice here, so suppress the warning
```

Comment:
Just tested it, and you're correct here. I've switched to just setting the profile to state it doesn't have hard edges, with a comment to note it might have to be changed if it's made x-analytic in the future